### ISSUE TITLE
Complete Tensorflow support without installing Keras separately

### ISSUE THREAD

**Issue by henrysky** [henrysky/astroNN#4]
### System information
- **Have I written custom code?**: Irrelevant
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04 or Windows 10 v1709 x64)**: Irrelevant
- **astroNN (Build or Version)**: Irrelevant
- **Did you try the latest astroNN commit?**: Irrelevant
- **TensorFlow installed from (source or binary, official build?)**: Irrelevant
- **TensorFlow version**: >=1.5.0
- **Keras version**: Irrelevant
- **Python version**:  >=3.5
- **CUDA/cuDNN version (Only neccessary if you are using Tensorflow-gpu)**: Irrelevant
- **GPU model and memory (Only neccessary if you are using Tensorflow-gpu)**: Irrelevant
- **Exact command/script to reproduce (optional)**: Irrelevant

### Describe the problem
Since Tensorflow 1.5.0, Keras is an official part of Tensroflow API (tensorflow.keras). astroNN should support both `keras` and `tensorflow.keras`.

What is done?
- Loss functions are all written in `tensorflow`

What is not done?
- Layers and CallBacks are all written with `keras`
- Models and training process are all written with `keras`
- Session management is currently done with `keras`
- astroNN's configuration file

### Source code / logs

A relevant discussion on [Keras github](https://github.com/keras-team/keras/issues/5299#issuecomment-278166426)

### Suggestion

- Configuration file (let users choose `keras` or `tensorflow.keras`)
- Default confuguration should point to `keras` or `tensorflow.keras`??

**Comment by henrysky** [comment]
`astroNN` basically supports `tensorflow` without the need of installing `keras` by using `tensorflow.keras` for now

Using `keras` is still recommended. The issue remains open because `tensorflow.keras` needs more tetsing

**Comment by henrysky** [comment]
an issue encountered, I have reported in tensorflow/tensorflow#17419

**Comment by henrysky** [comment]
``plot_model()`` is currently broken in ``tf.keras`` -> tensorflow/tensorflow#17633

I will close this astroNN issue as most of the tensorflow functionality is validated to work properly
